lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+          loaded_spec_files = RSpec.configuration.loaded_spec_files
+
+          if loaded_spec_files.include?(new_definition) && old_definition == new_definition
+            RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
+              |WARNING: Shared example group '#{key}' has been previously defined at:
+              |  #{old_definition}
+              |...and you are now defining it at:
+              |  #{new_definition}
+              |The new definition will overwrite the original one.
```

Comment:
I think in this case, since we know the old and new definition locations are the same, it makes sense to word this paragraph differently. Instead of printing off the same location twice, state that it is defined at `<old or new location>` and the file is being loaded a second time.